packages/api/actor/src/route/actors.rs

```diff
@@ -439,6 +440,98 @@ pub async fn upgrade(
 	Ok(json!({}))
 }
 
+// MARK: POST /actors/upgrade
+pub async fn upgrade_all(
+	ctx: Ctx<Auth>,
+	body: models::ActorUpgradeAllActorsRequest,
+	query: GlobalQuery,
+) -> GlobalResult<models::ActorUpgradeAllActorsResponse> {
+	let CheckOutput { env_id, .. } = ctx.auth().check(ctx.op_ctx(), &query, false).await?;
+
+	let tags = unwrap_with!(body.tags, API_BAD_BODY, error = "missing property `tags`");
+
+	ensure_with!(
+		tags.as_object().map(|x| x.len()).unwrap_or_default() <= 64,
+		API_BAD_BODY,
+		error = "Too many tags (max 64)."
+	);
+
+	let tags = unwrap_with!(
+		serde_json::from_value::<HashMap<String, String>>(tags).ok(),
+		API_BAD_BODY,
+		error = "`tags` must be `Map<String, String>`"
+	);
+
+	for (k, v) in &tags {
+		ensure_with!(
+			k.len() <= 256,
+			API_BAD_BODY,
+			error = format!(
+				"tags[{:?}]: Tag label too large (max 256 bytes).",
+				&k[..256]
+			),
+		);
+
+		ensure_with!(
+			v.len() <= 1024,
+			API_BAD_BODY,
+			error = format!("tags[{k:?}]: Tag value too large (max 1024 bytes)."),
+		);
+	}
+
+	let build_id = resolve_build_id(&ctx, env_id, body.build, body.build_tags.flatten()).await?;
+
+	// Work in batches
+	let mut count = 0;
+	let mut cursor = None;
+	loop {
+		let list_res = ctx
```

Comment:
this is going to be problematic, it's very likely that applications will have > 10,000 actors. we need to stream the actors with a cursor. i'm fine with adding a ticket to do this post-mvp as long as it has a due date.

Reply:
This is already chunking the actors with a cursor, its just doing so in chunks of 10k

Reply:
i'm blind